routers/user.js

```diff
@@ -35,30 +35,78 @@ class User {
                 lastName: context.userInfo.lastName,
                 createdAt: session_currentTime,
                 updateAt: session_currentTime
-                }
+            }
             await this.userCollection.insert(user);
+        } else {
+            user = result[0];
         }
-        else{
-            user =  result[0];
-        }
-        if (result.matchedCount < 1){
+        if (result.matchedCount < 1) {
             let error = "there is an error getting the result";
             console.error(error)
             throw new Error(error)
-          } 
+        }
         context.userInfo = {
             ...user,
             ...context.userInfo
         }
         return user
     }
-    
 
-    async updateMyUser(parm, context) {
+    async updateMyUser(params, context) {
```

Comment:
Verify that the userInfo in the session has been initialized before doing anything. This can be checked by making sure that context.userInfo._id is not null. If it is null, then throw an error message saying that the user data in the session has not been initialized